src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -1,90 +1,67 @@
-        return null;
+        return elements.stream()
+                .findFirst();
     }
 
-    /**
-     * <p>4. Дана коллекция строк List of String elements
-     * (приметр: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Найти элемент в коллекции равный `element` или кинуть ошибку NoSuchElementException</p>
-     **/
     public String findElement(List<String> elements, String element) {
-        return null;
+        return elements.stream()
+               .filter(item -> item.equals(element))
+               .findFirst()
+               .orElseThrow();
     }
 
-    /**
-     * <p>5. Дана коллекция чисел List of Integer numbers (пример: Arrays.asList(6, 2, 3, 7, 2, 5))
-     * Отнимите от каждого элемента, который стоит на непарной позиции (имеет не парный индекс) 1
-     * и верните среднее арифметическое всех нечетных чисел или киньте ошибку
-     * NoSuchElementException</p>
-     **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        return IntStream.range(0, numbers.size())
+                .map(x -> x % 2 == 0 ? numbers.get(x) : numbers.get(x) - 1)
+                .filter(x -> x % 2 == 1)
+                .average()
+                .orElseThrow();
     }
 
-    /**
-     * <p>6. Дана коллекция класс People (с полями name — имя, age — возраст, sex — пол),
-     * вида Arrays.asList( new People(«Вася», 16, Sex.MAN),
-     * new People(«Петя», 23, Sex.MAN),
-     * new People(«Елена», 42, Sex.WOMEN),
-     * new People(«Иван Иванович», 69, Sex.MAN)).
-     * Задача: Выбрать мужчин-военнообязанных (от `fromAge` до `toAge` лет)</p>
-     **/
     public List<People> manSelectByAge(List<People> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(people -> people.getSex().equals(People.Sex.MAN)
+                        && people.getAge() >= fromAge
+                        && people.getAge() <= toAge)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * <p>6. Дана коллекция класс People (с полями name — имя, age — возраст, sex — пол),
-     * вида Arrays.asList( new People(«Вася», 16, Sex.MAN),
-     * new People(«Петя», 23, Sex.MAN),
-     * new People(«Елена», 42, Sex.WOMEN),
-     * new People(«Иван Иванович», 69, Sex.MAN)).
-     * Задача: Найти всех потенциально работоспособных людей в выборке от fromAge до
-     * femaleToAge для женщин, и от fromAge до maleToAge для мужчин
-     * Пример: от 18 лет и учитывая что женщины выходят в 55 лет, а мужчина в 60</p>
-     **/
     public List<People> workablePeople(int fromAge, int femaleToAge,
                                        int maleToAge, List<People> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(people -> people.getSex().equals(People.Sex.WOMEN)
+                        ? people.getAge() >= fromAge && people.getAge() <= femaleToAge
```

Comment:
`people.getAge() >= fromAge` is common to male and female.
try to evoid code duplication